zoo/Main.java

```diff
@@ -1,53 +1,110 @@
+        }
+
+
+        Visitor visitor = new Visitor();
+        try {
+            visitor.setName("Victor");
+        } catch(nameException e) {
 
-        Tiger tiger = new Tiger();
-        tiger.setSex("Male");
+            LOGGER.error("Name exception occurred: " + e.getMessage());
+        }
+
+        Ticket ticket = new Ticket();
+        ticket.setTicketID(1);
+        ticket.setCost(10);
+
+        try {
+            ticket.setDate(11, 16, 2023);
+        } catch(ExpiredTicketException e) {
+
+            LOGGER.error("ExpiredTicketException occurred: " + e.getMessage());
+        }
+
+        visitor.setTicket(ticket);
+        myZoo.addVisitor(visitor);
+
+        welcomeVisitor();
+
+        myZoo.addLocation("Tiger's Den");
+        myZoo.addLocation("Eagle's cage");
+        myZoo.addLocation("Crocodile pool");
+
+        Tiger tiger = new Tiger(myZoo);
+        tiger.setSex(Sex.MALE);
         myZoo.addAnimal(tiger);
         System.out.println(tiger);
 
-        BaldEagle eagle = new BaldEagle();
-        eagle.setSex("Female");
+        BaldEagle eagle = new BaldEagle(myZoo);
+        eagle.setSex(Sex.FEMALE);
         eagle.setCanFly(true);
         myZoo.addAnimal(eagle);
         System.out.println(eagle);
 
-        Crocodile croc = new Crocodile();
-        croc.setSex("Male");
+        Crocodile croc = new Crocodile(myZoo);
+        croc.setSex(Sex.MALE);
         croc.setIsVenomous(false);
         myZoo.addAnimal(croc);
         System.out.println(croc);
 
-        Visitor visitor = new Visitor();
-        visitor.setName("Victor");
-        Ticket ticket = new Ticket();
-        ticket.setTicketID(1);
-        visitor.setTicket(ticket);
-        myZoo.addVisitor(visitor);
-
         ZooKeeper zooKeeper = new ZooKeeper();
-        zooKeeper.setName("Pieter");
+        try {
+            zooKeeper.setName("Pieter");
+        } catch(nameException e) {
+
+            LOGGER.error("NameException occurred: " + e.getMessage());
+        }
+
         zooKeeper.setNumFood(5);
         myZoo.addZooKeeper(zooKeeper);
 
+        try {
+            visitor.setLocation("Tiger's Den", myZoo);
+            tiger.setLocation("Tiger's Den", myZoo);
+            zooKeeper.setLocation("Tiger's Den", myZoo);
+        } catch(LocationException e) {
+
+            LOGGER.error("LocationException occurred: " + e.getMessage());
+        }
+
         System.out.println("Welcome to " + myZoo.getName());
         System.out.println("We have " + myZoo.getAnimals().size() + " animals.");
 
-        myZoo.addLocation("Tiger's Den");
-        visitor.setLocation("Tiger's Den");
-        tiger.setLocation("Tiger's Den");
-        zooKeeper.setLocation("Tiger's Den");
+        tiger.showSpecies();
+        try{
+            zooKeeper.feedAnimal(tiger);
+        } catch(FeedAnimalException e) {
+
+            LOGGER.error("FeedAnimalException occurred: "+ e.getMessage());
+        }
```

Comment:
please fix everywhere the same issue
```suggestion
        } catch(FeedAnimalException e) {
            LOGGER.error("FeedAnimalException occurred: "+ e.getMessage());
        }
```